### PR TITLE
Adding styling and a component for Foundation-style modal popups

### PR DESCRIPTION
**Pull request by geetotes**

### CODE REVIEW

File: app/assets/javascripts/revelry/examples/Modals.js.cjsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Rev.pullInto(this) 'Lipsum', 'Modal'
+
+Button = Rev.Components.Button
```
Comment (jwietelmann) [inline app/assets/javascripts/revelry/examples/Modals.js.cjsx]: add `'Button'` to the `pullInto` line

File: app/assets/stylesheets/revelry/ui/modals/Modal.scss [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+.RevModalBgShow {
+  display: block;
+  visibility: visible;
+}
+.RevModalReveal {
+  display: block;
+  visibility: visible;
+  opacity: 1;
+  top: 0px;
+}
```
Comment (jwietelmann) [inline app/assets/stylesheets/revelry/ui/modals/Modal.scss]: Follow SUIT conventions. `RevModal` is the component. Then there are subcomponents and modifiers.

File: app/assets/javascripts/revelry/ui/modals/Modal.js.cjsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  modalStyle: ->
+    top: window.scrollY + "px"
+
+  render: ->
+    if @state.show
+      <div>
+        <div className="reveal-modal-bg RevModalBg RevModalBgShow" onClick={ @handleBackgroundClick }></div>
+        <div className="active reveal-modal RevModalReveal" style={ @modalStyle() }>
+          <div className="modal-content">
```
Comment (jwietelmann) [inline app/assets/javascripts/revelry/ui/modals/Modal.js.cjsx]: What's the deal with all these extra classes you didn't write CSS rules for?

File: app/assets/javascripts/revelry/ui/modals/Modal.js.cjsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    @setState show: nextProps.show
+
+  componentWillUnmount: ->
+    @stopListening()
+
+  getDefaultProps: ->
+    show: false
+
+  getInitialState: ->
+    show: @props.show
```
Comment (jwietelmann) [inline app/assets/javascripts/revelry/ui/modals/Modal.js.cjsx]: You're creating an unholy tangle of props and state and creating two sources of truth that require perfect synchronicity between the component and its parent to not blow up. For this reason, I'm closing the PR for now. The modal should probably not manage its own state. Its parent should either render it or not. And if you want transitions, read https://facebook.github.io/react/docs/animation.html

File: app/assets/javascripts/revelry/ui/modals/Modal.js.cjsx (hunk 3 above)
Comment (geetotes) [inline app/assets/javascripts/revelry/ui/modals/Modal.js.cjsx]: `reveal-modal-bg`, `active`, and `reveal-modal` are part of Foundation. I'll pull out `modal-content`.

File: app/assets/javascripts/revelry/ui/modals/Modal.js.cjsx (hunk 4 above)
Comment (geetotes) [inline app/assets/javascripts/revelry/ui/modals/Modal.js.cjsx]: So if the parent renders it or not, how does the parent know when the background is clicked to close out the modal?

**Pull request closed without merging**